packages/flutter_test/lib/src/finders.dart

```diff
@@ -583,3 +606,37 @@ class _DescendantFinder extends Finder {
     return candidates;
   }
 }
+
+class _AncestorFinder extends Finder {
+  _AncestorFinder(this.descendant, this.ancestor, { this.matchRoot: false }) : super(skipOffstage: false);
+
+  final Finder ancestor;
+  final Finder descendant;
+  final bool matchRoot;
+
+  @override
+  String get description {
+    return '${ancestor.description} which is an ancestor of ${descendant.description}';
```

Comment:
super nit: maybe keep the matchRoot branch from _DescendantFinder

Reply:
I added a matchRoot description.